### PR TITLE
Improve default locale path discovering.

### PR DESCRIPTION
**Pull request by mondeja**
### Changes proposed in this pull request:

* Default locale path only is computed when `humanize.i18n.activate` function is called.
* Handle `__file__ is None` and `__file__` not defined cases.  
* Add tests cases increasing code coverage.

### CODE REVIEW

File: src/humanize/i18n.py [hunk 1]
```diff
@@ -7,11 +7,14 @@
...
 
-try:
-    _DEFAULT_LOCALE_PATH = os.path.join(os.path.dirname(__file__), "locale")
-except AttributeError:
-    # in case that __file__ does not exist
-    _DEFAULT_LOCALE_PATH = None
+
+def get_default_locale_path():
+    if __file__ is None:
+        return
```
Comment (hugovk) [inline src/humanize/i18n.py]: Let's explicitly return `None`
```suggestion
        return None
```

File: src/humanize/i18n.py [hunk 2]
```diff
@@ -7,11 +7,14 @@
...
+    try:
+        return os.path.join(os.path.dirname(__file__), "locale")
+    except NameError:
+        return
```
Comment (hugovk) [inline src/humanize/i18n.py:17]:
```suggestion
        return None
```

File: tests/test_i18n.py [hunk 3]
```diff
@@ -17,3 +19,38 @@ def test_i18n():
         humanize.i18n.deactivate()
         assert humanize.naturaltime(three_seconds) == "3 seconds ago"
         assert humanize.ordinal(5) == "5th"
+
+
+def test_i18n_default_locale_path():
```
Comment (hugovk) [inline tests/test_i18n.py:24]: Please can you split this into three separate functions, one for defined, one for `None` and one for not defined?

This'll ensure one thing is tested at a time; that the later parts of the test does not depend on the earlier parts.

File: tests/test_i18n.py [hunk 4]
```diff
@@ -17,3 +19,38 @@ def test_i18n():
...
+    assert i18n.get_default_locale_path() is None
+
+    # __file__ not defined
+    del i18n.__file__
+    with pytest.raises(NameError) as excinfo:
+        i18n.get_default_locale_path()
+    assert str(excinfo.value) == "name '__file__' is not defined"
+
+
+def test_activate_without_default_locale_path():
```
Comment (hugovk) [inline tests/test_i18n.py:38]: And likewise split this into two.

**Comment by mondeja** [comment]
All resolved, seems that coverage upload is failing...

**Comment by hugovk** [comment]
Thanks! Yeah, the coverage upload has been a bit flaky of late, nothing to worry about here. I've restarted the CI (but it might fail again 🙃).

File: src/humanize/i18n.py [hunk 5]
```diff
@@ -7,11 +7,14 @@
 _TRANSLATIONS = {None: gettext_module.NullTranslations()}
 _CURRENT = local()
 
-try:
-    _DEFAULT_LOCALE_PATH = os.path.join(os.path.dirname(__file__), "locale")
-except AttributeError:
-    # in case that __file__ does not exist
-    _DEFAULT_LOCALE_PATH = None
+
+def get_default_locale_path():
```
Comment (hugovk) [inline src/humanize/i18n.py:11]: One last thought, should this be marked as an internal function or is it something we want to expose (and document via a docstring) via the API?

```suggestion
def _get_default_locale_path():
```

File: src/humanize/i18n.py [hunk 6]
```diff
@@ -26,7 +29,7 @@ def activate(locale, path=None):
     @param locale: language name, eg 'en_GB'
     @param path: path to search for locales"""
     if path is None:
-        path = _DEFAULT_LOCALE_PATH
+        path = get_default_locale_path()
```
Comment (hugovk) [inline src/humanize/i18n.py:32]:
```suggestion
        path = _get_default_locale_path()
```

File: tests/test_i18n.py [hunk 7]
```diff
@@ -17,3 +19,43 @@ def test_i18n():
         humanize.i18n.deactivate()
         assert humanize.naturaltime(three_seconds) == "3 seconds ago"
         assert humanize.ordinal(5) == "5th"
+
+
+def test_default_locale_path_defined__file__():
+    i18n = importlib.import_module("humanize.i18n")
+    assert i18n.get_default_locale_path() is not None
```
Comment (hugovk) [inline tests/test_i18n.py:26]:
```suggestion
    assert i18n._get_default_locale_path() is not None
```

File: tests/test_i18n.py [hunk 8]
```diff
@@ -17,3 +19,43 @@ def test_i18n():
...
+
+
+def test_default_locale_path_null__file__():
+    i18n = importlib.import_module("humanize.i18n")
+    i18n.__file__ = None
+    assert i18n.get_default_locale_path() is None
```
Comment (hugovk) [inline tests/test_i18n.py:32]:
```suggestion
    assert i18n._get_default_locale_path() is None
```

File: tests/test_i18n.py [hunk 9]
```diff
@@ -17,3 +19,43 @@ def test_i18n():
...
+
+
+def test_default_locale_path_undefined__file__():
+    i18n = importlib.import_module("humanize.i18n")
+    del i18n.__file__
+    i18n.get_default_locale_path() is None
```
Comment (hugovk) [inline tests/test_i18n.py:38]:
```suggestion
    i18n._get_default_locale_path() is None
```

File: src/humanize/i18n.py (hunk 5 above)
Comment (mondeja) [reply src/humanize/i18n.py]: Resolved marking as internal function.

**Comment by hugovk** [comment]
Thank you!

**Pull request merged**